File: dotnet/DesignPatterns/StructuralDesignPatterns/AdapterDesignPattern/AdapterDesignPattern.Example1/Adapters/PaypalAdapter.cs
using AdapterDesignPattern.Example1.Interfaces;
using AdapterDesignPattern.Example1.Services;

namespace AdapterDesignPattern.Example1.Adapters;

public class PaypalAdapter : IPaymentProcessor
{
	private readonly PaypalSDK _paypalSDK;

	public PaypalAdapter(PaypalSDK paypalSDK)
	{
		_paypalSDK = paypalSDK;
	}

	public void ProcessPayment(double amount, string currency)
	{
		_paypalSDK.MakePayment(amount.ToString(), currency);
	}
}

File: nodejs/DesignPatterns/StructuralDesignPatterns/AdapterDesignPattern/AdapterDesignPattern.Example1/src/adapters/PaypalAdapter.ts
import { IPaymentProcessor } from '../interfaces/IPaymentProcessor';
import { PaypalSDK } from '../services/PaypalSDK';

export class PaypalAdapter implements IPaymentProcessor {
  constructor(private readonly paypalSDK: PaypalSDK) {}

  processPayment(amount: number, currency: string): void {
    this.paypalSDK.makePayment(amount.toString(), currency);
  }
}
